llvm-external-projects/iree-dialects/lib/Dialect/IREEPyDM/Transforms/ToIREE/LoweringPatterns.cpp

```diff
@@ -41,15 +38,22 @@ enum class ExceptionCode : int {
 
 }  // namespace
 
+static Type getVariantListType(Builder &builder) {
+  return builder.getType<iree::ListType>(builder.getType<iree::VariantType>());
+}
+
 static Value getNullValue(Location loc, OpBuilder &builder, Type t) {
   return TypeSwitch<Type, Value>(t)
-      .Case<iree_d::ListType>([&](auto t) -> Value {
-        return builder.create<iree_d::NullOp>(loc, t);
+      .Case<iree::ListType>([&](auto t) -> Value {
+        // TODO: If it becomes important to optimize this, come up with a way
+        // to return an empty list without creating one.
+        return builder.create<iree::ListCreateOp>(
```

Comment:
I do have a util.null now that's used internally for this kind of stuff - may be worth adding to the iree dialect too.

Reply:
It is there and I switched to this because it wasn't working (and I convinced myself that it couldn't work for this case vs just being broken). But leaving the TODO to resolve at some point.